File: tablix/tablix_support/report/issue_report_boq/issue_report_boq.js
// Copyright (c) 2016, Tablix and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Issue Report BOQ"] = {
	"filters": [
		{fieldname: "project_site", label: __("Project Site"), fieldtype:"Link", options: "Maintenance Contract"},
		{fieldname: "status", label: __("Issue Status"), fieldtype:"Select", options: "\nOpen\nClosed", default:"Closed"},
		{fieldname: "report_type", label: __("Report Type"), fieldtype: "Select", options: "Yearly\n"},
		{fieldname: "fiscal_year", label: __("Fiscal Year"), fieldtype:"Link", options:"Fiscal Year", default:frappe.boot.sysdefaults.fiscal_year}
		
	]
}

File: tablix/tablix_support/report/issue_report_boq/issue_report_boq.py
# Copyright (c) 2013, Tablix and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe import _
from tablix.tdashboard.page.issue_dashboard.issue_dashboard import get_data

def execute(filters=None):
	columns, data = [], []
	data = _get_data(filters)
	columns = get_columns()
	return columns, data



def get_columns():
	
	return [
		_("Issue Name")+":Link/Issue:100", _("Project Site")+ ":Link/Maintenance Contract:150",
		_("Failure Category")+":Data:150", _("Subject")+":Data:250",
		_("Opening Datetime")+ ":Data:175", _("Close Datetime")+ ":Data:175",
		_("Attended Datetime")+ ":Data:150", _("Rectification Datetime")+":Data:150",
		_("Open-Close-Diff")+":Data:150", _("Attend-Rectify-Diff")+":Data:150",
		
	]


def _get_data(filters):
		
	data = []
	for item in get_data(filters):
		data.append([
			item.get("name"), item.get("proj_site"), item.get("failure_cat"),
			item.get("subject"), item.get("opening_datetime"), item.get("modified"),
			item.get("attended_datetime"), item.get("rectification_datetime"),
			item.get("open_close_time"), item.get("rectified_time")
		])		

	return data	
